### PR TITLE
[mapbox-gl] Add type definition for resetNorthPitch

### PR DESCRIPTION
**Pull request by michael-mml**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://docs.mapbox.com/mapbox-gl-js/api/map/#map#resetnorthpitch>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by dmytro-gokun** [comment]
@michael-mml Thanks a lot for your submission. I only have one question: why are there so many formatting changes unrelated at the actual functionality change. Did you use prettier config included with DefinitelyTyped?

File: types/mapbox-gl/index.d.ts [hunk 1]
```diff
@@ -421,6 +421,8 @@ declare namespace mapboxgl {
 
         resetNorth(options?: mapboxgl.AnimationOptions, eventData?: mapboxgl.EventData): this;
 
+        resetNorthPitch(options?: mapboxgl.AnimationOptions, eventData?: mapboxgl.EventData): this;
```
Comment (dmytro-gokun) [inline types/mapbox-gl/index.d.ts:424]: Small thing here: "?" in TS means "T | undefined", but in Flow (which is used by mapbox) it means "T | undefined | null", so we should reflect that fact. I know, it has not been very consistent across this @types/mapbox-gl project from the beginning, but let's try to fix this over time.

**Review by dmytro-gokun — CHANGES_REQUESTED** [review]

**Comment by michael-mml** [comment]
> 
> 
> @michael-mml Thanks a lot for your submission. I only have one question: why are there so many formatting changes unrelated at the actual functionality change. Did you use prettier config included with DefinitelyTyped?

Hi @dmytro-gokun , I ran `npm run lint mapbox-gl` as suggested. Should I be using `prettier` instead?

File: types/mapbox-gl/index.d.ts (hunk 1 above)
Comment (michael-mml) [reply types/mapbox-gl/index.d.ts:424]: What would be a more reasonable approach:
1. adding `| null` i.e. `options?: mapboxgl.AnimationOptions | null`
or
2. creating some sort of new type that allows nullable i.e. `export type NullableAnimationOptions = mapboxgl.AnimationOptions | null;`

Comment (dmytro-gokun) [reply types/mapbox-gl/index.d.ts:424]: I'd go for #1. That's what already been done in several places, I believe.

**Comment by dmytro-gokun** [comment]
@michael-mml Those are not mutually exclusive. Lint is used to check for possible errors. And prettier is used to format code uniformely. You need to configure your IDE so that it uses prettier config from the root of this project.

**Review by dmytro-gokun — APPROVED** [review]

**Comment by dmytro-gokun** [comment]
Ready to merge

**Pull request merged**